torch/csrc/jit/passes/peephole.cpp

```diff
@@ -39,12 +41,15 @@ void PeepholeOptimize(Block * block) {
               input_node->destroy();
             }
           }
+          destroyed = true;
         }
         break;
     }
 
-    for (Block * sub_block : n->blocks()) {
-      PeepholeOptimize(sub_block);
+    if (!destroyed) {
+        for (Block * sub_block : n->blocks()) {
+            PeepholeOptimize(sub_block);
+        }
```

Comment:
Couldn’t we just move this loop before the switch? The order doesn’t really matter here, and that’s less bookkeeping

Reply:
@apaszke ok done